docs/reference/python/conventions.md

```diff
@@ -0,0 +1,162 @@
+process, we use [autopep8](https://github.com/hhatto/autopep8), which is a tool that resolves most style issues.
+However, there will be some issues that must be resolved by you!
+
+!!! note
+
+    Our CI automatically checks that your code follows the pep8 standard. If it does not, your pull requests will
+    be rejected until those issues are resolved!
+
+### Installation
+
+To install the style guide tools, use the `pip` package manager! If you use a virtual environment, you will have to
+install it there as well.
+
+```bash title="Installing the style guide tools"
+# Install flake8
+pip install flake8
+
+# Install autopep8
+pip install --upgrade autopep8
+```
+
+### Usage
+
+Use the following to utilize the style guide tools to automatically enforce the style guidelines. Remember
+to do this during development before you commit your code!
+
+```bash title="Using the style guide tools"
+# Automatically resolve style issues
+autopep8 --in-place --aggressive --max-line-length 120 --aggressive <PATH TO FILE>
+
+# Get the statistics on what still needs to be resolved
+flake8 --statistics --max-line-length 120 <PATH TO FILE>
+```
+
+## Documentation
+
+Code is written once and read a thousand times, so it is important to provide good documentation for current
+and future members of the software team. The major things that we document in our code are:
+
+1. **Classes and Objects:**
+    - What does it represent? What is it used for?
+    - What are its member variables? What are they used for?
+2. **Functions:**
+    - What are the inputs and outputs?
+    - What is the overall behavior and purpose of the function?
+3. **Code:**
+    - Is a line of code obscure and/or not clear? Add an inline comment to clear things up.
+    - Break down a large process.
+
+Ideally, the third point should be avoided as much as possible since we would want our code to be
+self explanatory. It should be done only when absolutely necessary.
+
+### Generating docstrings
+
+We use a vscode extension called [autoDocstring](https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring)
+which autogenerates docstrings that we use to document our code. To install this extension, go to the `Extensions` tab in
+vscode and search `autoDocstring` in the marketplace.
+
+To generate docstrings, type `"""` or `'''` at the beginning of the function that you want to document and the template
+will be generated for you!
+
+!!! note
+
+    The autoDocstring extension only works for functions. It does not work for classes and objects, so documenting these
+    will have to be done manually. Be sure to follow the same format used by functions.
+
+### Example on documentation
+
+It's hard to imagine what good documentation looks like. We provide a few examples below of documenting code using the
+autoDocstring extension.
+
+```python title="Documentation example on a function"
+def inner_product(v1, v2):
```

Comment:
We should discuss whether we want to enforce using type hints. They work well with autoDocstring, but I feel like they are a bit restrictive

Reply:
We talked about this already. You said that we should avoid them, but I originally thought they were a good idea.